nexus/src/app/background/init.rs

```diff
@@ -485,54 +644,27 @@ impl BackgroundTasks {
             Box::new(abandoned_vmm_reaper::AbandonedVmmReaper::new(datastore)),
             opctx.child(BTreeMap::new()),
             vec![],
-        );
-
-        BackgroundTasks {
-            driver,
-            task_internal_dns_config,
-            task_internal_dns_servers,
-            task_external_dns_config,
-            task_external_dns_servers,
-            task_metrics_producer_gc,
-            task_external_endpoints,
-            external_endpoints,
-            task_nat_cleanup,
-            task_bfd_manager,
-            task_inventory_collection,
-            task_physical_disk_adoption,
-            task_phantom_disks,
-            task_blueprint_loader,
-            task_blueprint_executor,
-            task_crdb_node_id_collector,
-            task_service_zone_nat_tracker,
-            task_switch_port_settings_manager,
-            task_v2p_manager,
-            task_region_replacement,
-            task_region_replacement_driver,
-            task_instance_watcher,
-            task_service_firewall_propagation,
             task_abandoned_vmm_reaper,
-            task_vpc_route_manager,
-        }
-    }
+        );
 
-    /// Activate the specified background task
-    ///
-    /// If the task is currently running, it will be activated again when it
-    /// finishes.
-    pub fn activate(&self, task: &TaskHandle) {
-        self.driver.activate(task);
+        driver
     }
 }
 
+/// Starts the three DNS-propagation-related background tasks for either
+/// internal or external DNS (depending on the arguments)
+#[allow(clippy::too_many_arguments)]
 fn init_dns(
     driver: &mut Driver,
     opctx: &OpContext,
     datastore: Arc<DataStore>,
     dns_group: DnsGroup,
     resolver: internal_dns::resolver::Resolver,
     config: &DnsTasksConfig,
-) -> (TaskHandle, TaskHandle) {
+    task_config: &Activator,
+    task_servers: &Activator,
+    task_propagation: &Activator,
```

Comment:
I'm always going to call out cases where we take multiple positional arguments of the same type, since it's easy to swap them by accident. Since this isn't an API that's called externally to this module, it's not as big a concern: this is only called once, and the arguments are passed in the right order, and we don't really expect new calls to this function will be added. Adding an args struct so that we can name these arguments just for one call seems kinda like overkill.

However, I _can_ think of a pretty simple way to change this so we don't have to pass these as positionals that can be swapped, and also make the `clippy::too_many_arguments` go away: why not just change this function to a method on `&BackgroundTasks` (or pass a `&BackgroundTasks` argument if we don't want to re-indent the entire function in order to make it a method?). That way, this code can refer to the various activators by name, and the caller can't mix them up. I don't think this function also being able to touch the other activators is a problem, since it can just...not do that.

Reply:
The problem is that this function is called twice with two different sets of activators (internal DNS and then external DNS).  The first time, the caller uses `BackgroundTasks.task_internal_dns_config`, while the second time it uses `BackgroundTasks.task_external_dns_config`.  Same for the other two `Activator` arguments.  So I think these have to be parameters.